cloudinit/sources/DataSourceOracle.py

```diff
@@ -380,71 +398,148 @@ def _add_network_config_from_opc_imds(self, set_primary: bool = False):
-                    if vnic_dict.get("privateIp"):
-                        interface_config["addresses"].append(
-                            f"{vnic_dict['privateIp']}/{network.prefixlen}"
-                        )
-                    if vnic_dict.get("ipv6Addresses"):
-                        interface_config["addresses"].append(
-                            f"{vnic_dict['ipv6Addresses'][0]}/"
-                            f"{network.prefixlen}"
-                        )
-                self._network_config["ethernets"][name] = interface_config
+                    subnets = [{"type": "dhcp"}]
+            else:
+                subnets = []
+                if vnic_dict.get("privateIp"):
+                    subnets.append(
+                        {
+                            "type": "static",
+                            "address": (
+                                f"{vnic_dict['privateIp']}/"
+                                f"{network.prefixlen}"
+                            ),
+                        }
+                    )
+                if vnic_dict.get("ipv6Addresses"):
+                    subnets.append(
+                        {
+                            "type": "static",
+                            "address": (
+                                f"{vnic_dict['ipv6Addresses'][0]}/"
+                                f"{network.prefixlen}"
+                            ),
+                        }
+                    )
+            interface_config = {
+                "name": name,
+                "type": "physical",
+                "mac_address": mac_address,
+                "mtu": MTU,
+                "subnets": subnets,
+            }
+            self._network_config_v1["config"].append(interface_config)
+
+            ############################# V2 HERE #############################
+            # Why does this elif exist???
+            # Are there plans to switch to v2?
+            interface_config = {
+                "mtu": MTU,
+                "match": {"macaddress": mac_address},
+                "set-name": name,
+            }
+            self._network_config["ethernets"][name] = interface_config
+
+            if is_primary and is_ipv6_only:
+                interface_config["dhcp6"] = True
+            elif is_primary and not is_ipv6_only:
+                interface_config["dhcp4"] = True
+
+            if not is_primary:
+                interface_config["addresses"] = []
+                if vnic_dict.get("privateIp"):
+                    interface_config["addresses"].append(
+                        f"{vnic_dict['privateIp']}/{network.prefixlen}"
+                    )
+                if vnic_dict.get("ipv6Addresses"):
+                    interface_config["addresses"].append(
+                        f"{vnic_dict['ipv6Addresses'][0]}/"
+                        f"{network.prefixlen}"
+                    )
+            self._network_config["ethernets"][name] = interface_config
```

Comment:
I think everything up until here looks generally good. I haven't checked specific keys yet, but the general idea makes sense.

I think everything after here would make more sense in tests. It's good to make sure that our v2 rendering matches (or closely resembles :smile: ) our v1 rendering as you're doing, but we don't need to be converting or comparing them in production code.

Instead, I think it'd be most useful to have tests containing a few different realistic examples of networking data we get from Oracle, then take the rendered v2 generated here and compare that against  the rendered v1 from here, and feeding that through netplan to get a v2 to compare against our rendered v2 here.